src/libs/svg/lv_svg_decoder.c

```diff
@@ -0,0 +1,308 @@
+
+/*********************
+ *      DEFINES
+ *********************/
+
+#define DECODER_NAME    "SVG"
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static lv_result_t svg_decoder_info(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * src,
+                                    lv_image_header_t * header);
+static lv_result_t svg_decoder_open(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc);
+static void svg_decoder_close(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc);
+static uint8_t * alloc_file(const char * filename, uint32_t * size);
+static void svg_draw_buf_free(void * svg_buf);
+
+static void svg_draw(lv_layer_t * layer, const lv_image_decoder_dsc_t * dsc, const lv_area_t * coords,
+                     const lv_matrix_t * matrix);
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static struct _lv_draw_buf_handlers_t _svg_draw_buf_handler = {
+    .buf_free_cb = svg_draw_buf_free,
+};
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * Register the SVG decoder functions in LVGL
+ */
+void lv_svg_decoder_init(void)
+{
+    lv_image_decoder_t * dec = lv_image_decoder_create();
+    lv_image_decoder_set_info_cb(dec, svg_decoder_info);
+    lv_image_decoder_set_open_cb(dec, svg_decoder_open);
+    lv_image_decoder_set_close_cb(dec, svg_decoder_close);
+
+    dec->name = DECODER_NAME;
+}
+
+void lv_svg_decoder_deinit(void)
+{
+    lv_image_decoder_t * dec = NULL;
+    while((dec = lv_image_decoder_get_next(dec)) != NULL) {
+        if(dec->info_cb == svg_decoder_info) {
+            lv_image_decoder_delete(dec);
+            break;
+        }
+    }
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+static lv_result_t svg_decoder_info(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * src,
+                                    lv_image_header_t * header)
+{
+    lv_image_src_t src_type = src->src_type;
+
+    int width = 0;
+    int height = 0;
+
+    if(src_type == LV_IMAGE_SRC_FILE || src_type == LV_IMAGE_SRC_VARIABLE) {
+        const void * src_data = src->src;
+        uint8_t buf[16];
+
+        if(src_type == LV_IMAGE_SRC_FILE) {
+            /*Support only "*.svg" files*/
+            if(lv_strcmp(lv_fs_get_ext(src_data), "svg")) {
+                return LV_RESULT_INVALID;
+            }
+
+            uint32_t rn;
+            lv_fs_read(&src->file, buf, sizeof(buf), &rn);
+
+            if(rn != sizeof(buf)) return LV_RESULT_INVALID;
+
+            if(lv_memcmp(buf, "<?xml", 5) != 0
+               && lv_memcmp(buf, "<svg", 4) != 0)  {
+                return LV_RESULT_INVALID;
+            }
+
+            lv_display_t * disp = lv_display_get_default();
+            width = disp->hor_res;
+            height = disp->ver_res;
```

Comment:
> Why screen sized images?

Because SVG is a scalable image, its size cannot be known before decoding, and it may not be possible to set the size. So a default value is given